Temporary file will fail to clean up.

je3kmz opened the pull request:
When you run the test-mktemp script on FreeBSD, the mktemp(1) is incorrect.
Even if a temporary file is created by that, it does not delete it after it is unnecessary.

The 'mbmon' plug-in now can now use environment variables.

je3kmz committed:
Temporary file will fail to clean up.

je3kmz committed:
Command switch append

sumpfralle commented on plugins/node.d/mbmon_.in:
```diff
@@ -16,14 +16,24 @@
 what=$(basename "$0" | sed 's/^mbmon_//g')
-mbmon=${mbmon:-$(which mbmon)}
-
+mbmon="${mbmonpath:-$(which mbmon)} -c1 -r ${mbmonargs}"
```
Could you please elaborate a bit, how this changes the behaviour of the plugin for its current users?

Did everyone need to specify the `mbmon` variable as `/usr/local/bin/mbmon -c 1 -r`? Or did the previous version also work with `mbmon` being empty?
(I do not use this plugin, thus I have no idea, how people are using it)

How about keeping the previous `mbmon` definition and simply replace `-c 1 -r` in every `mbmon` call below with your new configurable `${mbmonargs}` variable?

je3kmz replied:
At least, in an environment where a plurality of sensors are attached, there is a case where the correct value is not detected.
In the case of one sensor, it works without problem scan to specify that it is operating even in the sky.

It's smarter to append '-c 1 -r' to the default of ${mbmonargs}.
I agree with the replacement.

sumpfralle commented on test-mktemp:
```diff
@@ -53,7 +53,7 @@ MKTEMP='mktemp -p /tmp/ $1'
 
 # FreeBSD way
-MKTEMP="mktemp -t /tmp"
+MKTEMP="mktemp -t $1"
```
Here `$1` would be resolved immediately (as an empty string). Did you intend this?

sumpfralle commented on test-mktemp:
```diff
@@ -37,7 +37,7 @@
 
 testfun () {
-	TEMP_FILE=`eval $MKTEMP 2>/dev/null`
+	TEMP_FILE=`eval $MKTEMP $1 2>/dev/null`
```
Do you know, if `$1` was missing by accident before?
How does this change the behaviour of the test?

je3kmz replied:
At that time, the mktemp command `-P' switch was not prepared, so we submitted a workaround, but now the problem has been solved.
The foundation is the source code [commit log](https://cgit.freebsd.org/src/commit/usr.bin/mktemp/mktemp/mktemp.c?id=AC6F924E1EBD983227E906EBA5EEDE).

Therefore, there is no need to check only FreeBSD.

sumpfralle reviewed (commented):
Thank you for your contribution!

I already commented on the `mbmon` changes separately.

Regarding the `mktemp` change: what exactly do you intend to improve with your change? How is the file removal related to this script? (I think, this needs to be handled by the caller)

je3kmz commented:
If mbmon runs without a switch,
In rare cases, the execution results may fluctuate in an environment where multiple sensors are installed.
It is also a workaround.

The change to mbmon-> mbmonpath has been adapted to other plugin descriptions.
If the change of the environment variable name is confusing, correct it only by adding a switch.

je3kmz committed:
Update test-mktemp

I'll fix it.
The fix on line 40 is a mistake.
The fix on line 56 did not require the -t switch.

Commit:
Fix: Can't open master/blib/libdoc/Munin::Master::HTMLOld.3pm

Commit:
Fix: FreeBSD ports skeleton

Commit:
REMOVE: dists/freebsd/munin-main

je3kmz commented:
This change proposal was rewritten to solve a problem when running a particular plug-in.

There was a problem running munin-amavis.
The MKTEMP environment variable absorbs OS dependent areas by the command column determined during installation.
This is because the command column generates multiple temporary files.
Because only one file is erased at the time of the cleanup at the end of the plug-in execution,
The remaining temporary files that have been generated will gradually remain.

munin-amavis
120line
TEMP_FILE=$(mktemp -t /tmp munin-amavis.XXXXXX)
In this way, a temporary file has been generated on the script.

$ /tmp//tmp.zb4Oabbs
$ munin-amavis.AjQkwW
Two files were made.

133line
 	/bin/rm -f "$TEMP_FILE"
The former file is not deleted during cleanup, and the latter is deleted.

test-mktemp
40lins
	 TEMP_FILE=`eval $MKTEMP $1 2>/dev/null`
This rewrite might have been unnecessary.


Lars Kruse wrote on 2019/08/22 5:27:
> sumpfralle commented on this pull request.
> 
> Thank you for your contribution!
> 
> I already commented on the `mbmon` changes separately.
> 
> Regarding the `mktemp` change: what exactly do you intend to improve with your change? How is the file removal related to this script? (I think, this needs to be handled by the caller)
> 
>> @@ -53,7 +53,7 @@ MKTEMP='mktemp -p /tmp/ $1'
>  testfun temp.XXXXXX
>  
>  # FreeBSD way
> -MKTEMP="mktemp -t /tmp"
> +MKTEMP="mktemp -t $1"
> 
> Here `$1` would be resolved immediately (as an empty string). Did you intend this?
> 
>> @@ -37,7 +37,7 @@
>  
>  
>  testfun () {
> -	TEMP_FILE=`eval $MKTEMP 2>/dev/null`
> +	TEMP_FILE=`eval $MKTEMP $1 2>/dev/null`
> 
> Do you know, if `$1` was missing by accident before?
> How does this change the behaviour of the test?
> 


--
The message uses bing translation in English and Japanese.

sumpfralle commented:
@je3kmz: sorry for letting your contribution rest for so long.

I have to admit, that I am a bit overwhelmed with the different topics you raised in this pull request.
Thus I just prepared a commit based on your change of the `mbmon` plugin (see 2a501095bc7b578ac4d80c09622e6d04105303c0). Thank you for preparing it.

Regarding your change to `test-mktemp`: could you please open another pull request and describe its purpose? I am a bit lost with this at the moment.

Regarding your improvements of the freebsd packagin: I am not sure, whether you really intended to add this amount of new files to the repository. Was this your intention? Just for comparison: the packaging for Debian and for other distributions happens in separate packaging repositories. But I do not know the usual workflow of freebsd. Thus please advise and ideally open another pull request.

Thank you!

sumpfralle commented on Makefile:
```diff
@@ -34,7 +34,7 @@ PODMAN8          := build/master/doc/munin-cron master/doc/munin master/doc/muni
 PYTHON_LINT_CALL ?= python3 -m flake8
 CONFVAR_SUBSTITUTION_FILES = \
-	master/blib/libdoc/Munin\:\:Master\:\:HTMLOld.3pm \
+	master/blib/libdoc/Munin\:\:Master\:\:HTMLOld.3 \
```
Could you please explain this change?

Here in my build environment the file really contains the extension `.3pm`. Is this different on your host?

je3kmz replied:
When building on FreeBSD, I get a message saying the file is missing.

```shell
wget https://github.com/munin-monitoring/munin/archive/refs/tags/2.0.76.tar.gz
bsdtar xf 2.0.76.tar.gz
cd munin-2.0.76
env LANG=C gmake DESTDIR=/tmp/munin PREFIX=/tmp/munin
```
..snip..
```log
        master/blib/libdoc/Munin\:\:Master\:\:HTMLOld.3pm master/blib/lib/Munin/Master/HTMLOld.pm node/blib/sbin/munin-node-configure node/blib/sbin/munin-node node/blib/sbin/munin-run build/doc/munin-node.conf.5
Can't open master/blib/libdoc/Munin::Master::HTMLOld.3pm: No such file or directory.
gmake[1]: Leaving directory '/tmp/munin-2.0.76'
touch build-confvar-substitution-stamp
```
